README.md

```diff
@@ -14,6 +14,16 @@ React component to load formsort flows.
 
 Embed flows within other webpages, allowing communication between the embed and the parent page.
 
+## Publishing
+
+This repo is published as a single unit with all packages at the same version. It uses [Craft](https://github.com/getsentry/craft) for publishing.
```

Comment:
You won't be publishing a new version of `web-embed-api` by its own anymore, you'll be publishing all the packages at once. Yes it may yield to bumps with no changes but it helps you to not hunt the matching versions for our packages. It also makes many things regarding publishing simpler.

If this becomes a problem we can revisit it in the future so I'd prefer to move forward with this unless you have strong objections.

Reply:
Is there no way for it to auto-detect which packages were changed and only publish those? That's what lerna used to do, so it feels a bit like a step back

Reply:
Craft doesn't deal with that. We can certainly do this, even use lerna for this part. The issue is, we'll be creating GitHub releases for each release and since this is a monorepo, it is easier for end users to track changes in each version change.

Reply:
```suggestion
The issue is, we'll be creating GitHub releases for each release and since this is a monorepo, it is easier for end users to track changes in each version change.
```
Do you mean it's easier if we _do_ only publish the changed packages, or if we publish and increase the version of each package every time? 

To me the idea of all packages pinned to the same version doesn't seem ideal. If I understand correctly, this would mean that a major version change to e.g. `web-embed-api` would also require the same version change to `custom-question-api`, even thought the two packages are independent of each other, and even if the latter had no changes to it.

Reply:
To me the idea of all packages being pinned to the same version doesn't seem ideal. If I understand correctly, this would mean that a major version change to, e.g. `custom-question-api`, would require to the make the same version change to `web-embed-api`, even though the two packages are completely independent of each other, and even if the latter had no changes to it.

Reply:
>If I understand correctly, this would mean that a major version change to, e.g. custom-question-api, would require to the make the same version change to web-embed-api, even though the two packages are completely independent of each other, and even if the latter had no changes to it.

That is true yes. My counter argument is that the scenario you provide seems unlikely for the current state of the repo: if there's a breaking change in one of these packages, it is very likely that either other packages will have a breaking change too (like an api change) or one of the packages they both depend on such as `@formsort/constants` has a breaking change.

Reply:
These are the current versions of the packages:
- constants: 1.6.0
- custom-question-api: 0.1.0
- eslint-config: 1.3.7
- tsconfig: 1.0.2
- web-embed-api: 2.2.0
- react-embed: 2.1.0

With this change, would we need to update all of these to be the same version (e.g. 2.2.1)?

Reply:
>With this change, would we need to update all of these to be the same version (e.g. 2.2.1)?

Yes, exactly. I think that's a good thing but since you raised the point and are concerned, I think we should find a [Stalemate Resolution Associate](http://www.portal2sounds.com/28). Pinging @fzembow and @atikenny for them to weigh in.

Reply:
Yup, I still have my concerns, but if @atikenny and @fzembow feel okay about this I'll be on board too

Reply:
I definitely do **not** want to be pushing major versions of sibling packages for unrelated reasons.... changelogs with major changes that have no notes associated with them? That's not a good developer experience.

Custom question API and web embed api have almost nothing to do with one another, besides perhaps their use of the same constants. The main reason these are in a monorepo in the first place is because of these more subtle relationships between the packages.

If the consequence of this is -- well, they should be in different repositories, not in a monorepo, then I'm fine with that, although that makes development more difficult.

Reply:
We can certainly separate the deploys while still in a monorepo with separate changelogs. The only issue is we won't be able to use GitHub Releases properly as they are scoped to the repo.

You can see some examples from here: https://github.com/getsentry/sentry-ruby/releases?page=5

Another downside is one having to manually set which repo they are releasing from the UI.

All that being said, since I am 2 to one, I'll yield and set this up for per-package deployment for now.